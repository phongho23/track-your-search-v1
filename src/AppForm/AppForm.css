.AppForm-form {
    background-color: lightpink;
    color: white;
    padding: 16px;
    max-width: 332px;
  }
  
  .AppForm-form label {
    display: block;
    font-size: 16px;
    padding-bottom: 8px;
  }
  
  .AppForm-form input,
  .AppForm-form textarea,
  .AppForm-form select {
    font-size: 14px;
    padding: 8px;
    width: 300px;
  }
  
  .AppForm-form select {
    height: 33px;
  }
  
  .AppForm-form > .field {
    margin-bottom: 16px;
  }
  .AppForm-form > .buttons {
    margin-top: 26px;
  }
  
  .AppForm-form button {
    background-color: blue;
    border-radius: 4px;
    border: 0;
    color: white;
    display: block;
    font-size: 18px;
    font-weight: bold;
    padding: 8px;
    text-align: center;
    text-decoration: none;
  }
  .AppForm-form button[type="submit"]:hover {
    background-color: lightblue;
    color: white;
  }
  