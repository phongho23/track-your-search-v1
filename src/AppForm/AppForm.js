import React from 'react'
import './AppForm.css'

export default function AppForm(props) {
  const { className, ...otherProps } = props
  return (
    <form
      className={['AppForm-Form', className].join(' ')}
      action='#'
      {...otherProps}
    />
  )
}
