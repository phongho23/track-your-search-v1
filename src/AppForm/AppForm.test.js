import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AppForm from './AppForm'

describe(`AppForm component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  }

  it('renders a form.AppForm by default', () => {
    const wrapper = shallow(<AppForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the AppForm given props', () => {
    const wrapper = shallow(<AppForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
