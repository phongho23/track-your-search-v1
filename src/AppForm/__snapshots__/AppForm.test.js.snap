// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppForm component renders a form.AppForm by default 1`] = `
<form
  action="#"
  className="AppForm-Form "
/>
`;

exports[`AppForm component renders the AppForm given props 1`] = `
<form
  action="#"
  className="AppForm-Form test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</form>
`;
